### PR TITLE
CLOUDP-167784: Migrate Global Clusters CLI stores to v2 sdk

### PR DESCRIPTION
**Pull request by swattyT**
## Proposed changes

_Jira ticket:_ CLOUDP-# [CLOUDP-167784](https://jira.mongodb.org/browse/CLOUDP-167784)

Closes #[issue number]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by gssbzn — DISMISSED** [review]
LGTM

File: internal/kubernetes/operator/deployment/deployment.go [hunk 1]
```diff
@@ -196,7 +196,7 @@ func buildGlobalDeployment(atlasRepSpec []*mongodbatlas.AdvancedReplicationSpec,
 			Db:                     ns.Db,
 			Collection:             ns.Collection,
 			CustomShardKey:         ns.CustomShardKey,
-			NumInitialChunks:       ns.NumInitialChunks,
+			NumInitialChunks:       int(*ns.NumInitialChunks),
```
Comment (wtrocki) [inline internal/kubernetes/operator/deployment/deployment.go:199]: It is recommended use getters from sdk and avoid using * to not get any memory allocation errors.

File: internal/kubernetes/operator/deployment/deployment_test.go [hunk 2]
```diff
@@ -232,7 +232,7 @@ func TestBuildAtlasAdvancedDeployment(t *testing.T) {
 							CustomShardKey:         managedNamespace[0].CustomShardKey,
 							IsCustomShardKeyHashed: managedNamespace[0].IsCustomShardKeyHashed,
 							IsShardKeyUnique:       managedNamespace[0].IsShardKeyUnique,
-							NumInitialChunks:       managedNamespace[0].NumInitialChunks,
+							NumInitialChunks:       int(*managedNamespace[0].NumInitialChunks),
```
Comment (wtrocki) [inline internal/kubernetes/operator/deployment/deployment_test.go:235]: same as above

**Review by wtrocki — APPROVED** [review]

File: internal/kubernetes/operator/deployment/deployment.go (hunk 1 above)
Comment (swattyT) [reply internal/kubernetes/operator/deployment/deployment.go:199]: Oh, will change this. Thanks Wojciech!

**Review by wtrocki — APPROVED** [review]

**Review by matt-condon — APPROVED** [review]
LGTM

**Pull request merged**